src/Compilers/Core/Portable/Diagnostic/DiagnosticInfo.cs

```diff
@@ -122,6 +122,25 @@ internal static void AssertMessageSerializable(object[] args)
             }
         }
 
+        [Conditional("DEBUG")]
+        internal static void AssertExpectedMessageArgumentsLength(CommonMessageProvider messageProvider, int errorCode, int actualLength)
+        {
+#if DEBUG
+            if (errorCode <= 0)
+            {
+                return;
+            }
```

Comment:
When can this happen?

Reply:
For `ErrorCode.Void` and `ErrorCode.Unknown` which are used as placeholders for errors.